polyply/src/load_library.py

```diff
@@ -101,12 +137,64 @@ def load_library(name, lib_names, extra_files):
     """
     force_field = vermouth.forcefield.ForceField(name)
     if lib_names and extra_files:
-        lib_files = _resolve_lib_paths(lib_names, DATA_PATH)
+        lib_files = _resolve_lib_paths(lib_names, DATA_PATH,
+                                       FORCE_FIELD_PARSERS.keys())
         all_files = lib_files + extra_files
     elif lib_names:
-        all_files = _resolve_lib_paths(lib_names, DATA_PATH)
+        all_files = _resolve_lib_paths(lib_names, DATA_PATH,
+                                       FORCE_FIELD_PARSERS.keys())
     elif extra_files:
         all_files = extra_files
```

Comment:
`all_files = _resolve_lib_paths(lib_names, DATA_PATH) + extra_files`
replaces this entire tree :)
Assuming of course that `lib_names` and `extra_files` are something sane such as empty lists when they're False-ish. And assuming that _resolve_lib_paths returns a list.

Reply:
Nice, that removed a lot of clutter!